demos/source/aws_iot_demo_jobs.c

```diff
@@ -0,0 +1,533 @@
+/*
+ * Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file aws_iot_demo_jobs.c
+ * @brief Demonstrates use of the AWS Jobs library.
+ */
+
+/* Config include. Should always come first per the style guide. */
+#include "iot_config.h"
+
+/* System Includes */
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/* Demo logging. */
+#include "iot_demo_logging.h"
+
+/* Platform Includes. */
+#include "platform/iot_clock.h"
+#include "platform/iot_network.h"
+#include "platform/iot_threads.h"
+
+/* MQTT Include. The underlying protocol for this demo. */
+#include "iot_mqtt.h"
+
+/* JSON Utilies for pulling out values. */
+#include "iot_json_utils.h"
+
+/* Jobs Library. */
+#include "aws_iot_jobs.h"
+#include "types/aws_iot_jobs_types.h"
```

Comment:
You don't need to include the types header, the library header already includes it.